rotkehlchen/tasks/calendar.py

```diff
@@ -91,10 +91,15 @@ def notify_reminders(
 
 
 def delete_past_calendar_entries(database: DBHandler) -> None:
-    """delete old calendar entries that the user has allowed to delete"""
+    """Delete past calendar events that are marked for auto-deletion,
+    but only if all associated reminders (if any) have been acknowledged."""
     now = ts_now()
     with database.conn.write_ctx() as write_cursor:
-        write_cursor.execute('DELETE FROM calendar WHERE timestamp < ? AND auto_delete=1', (now,))
+        write_cursor.execute(
+            'DELETE FROM calendar WHERE timestamp < ? AND auto_delete = 1 '
+            'AND identifier NOT IN (SELECT event_id FROM calendar_reminders WHERE is_acknowledged = 0);',  # noqa: E501
```

Comment:
```suggestion
            'AND identifier IN (SELECT event_id FROM calendar_reminders WHERE is_acknowledged = 1);',  # noqa: E501
```
maybe I'm missing something here, but wouldn't it be better to skip the `NOT` and simply check for `is_acknowledged = 1`?